main.go

```diff
@@ -15,16 +15,25 @@
 package main
 
 import (
+	"context"
 	"log"
 	"os"
+
+	compute "cloud.google.com/go/compute/apiv1"
 )
 
 func HostedInGAE() bool {
 	return os.Getenv("GAE_APPLICATION") != ""
 }
 
 func main() {
-	im := &PlaceholderIM{}
+	ctx := context.Background()
+	client, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+	im := NewGCPIM(client)
```

Comment:
encapsulate all this in a New....() function.

Reply:
done